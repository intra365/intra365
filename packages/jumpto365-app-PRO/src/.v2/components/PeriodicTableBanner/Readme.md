
# TODO: Make sample

# w 500
```js
<PeriodicTableBanner icon="" height="200" width="500" text="Office 365" bannerFontSize="11" />
```
# w 100
```js
<PeriodicTableBanner icon="" height="40" width="100" text="Office 365" bannerFontSize="2"/>
```

# w 1500
```js
<PeriodicTableBanner icon="" height="600" width="1500" text="Office 365" bannerFontSize="32"/>
```