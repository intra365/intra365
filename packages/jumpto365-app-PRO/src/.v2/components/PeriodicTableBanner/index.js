import React, { Component } from 'react';
import PropTypes from 'prop-types'
import jumpto365Logo from '../../media/Logo horizontal color - transparent background.png'


/**
 * Describe overall purpose of the component
 *
 * @export
 * @class PeriodicTableBanner
 * @extends {Component}
 */
export default class PeriodicTableBanner extends Component { 


    static propTypes  = {
        text : PropTypes.string,
        subtitle :PropTypes.string,
        logo : PropTypes.string,
        width : PropTypes.number,
        height : PropTypes.number,
        image: PropTypes.string
    }
    

    constructor(props) {
        super(props);
        this.state = {}
        this.topText = React.createRef();
    }

    componentDidMount = () => {
        
        this.update()
        
    }

    update = () => {
        
        if (this.props.image){
            this.setState({
            dynamic:false})
        }else
        {
        this.setState({fontSizeTop : parseInt(this.props.bannerFontSize*2),
            fontSizeHalfTop : parseInt(this.props.bannerFontSize),
            text:this.props.text ? this.props.text : "Office 365",
        dynamic:false})
    }

    }
    componentDidUpdate = (previousProps, previousState) => {
        const node = this.topText.current;
        if ((previousProps.width !== this.props.width) || (previousProps.text !== this.props.text) || (previousProps !== this.props)  ){
            this.update()
        }
   

      }

    /**
     * Required method return the output of the component
     *
     * @returns
     * @memberof PeriodicTableBanner
     */
    render() {  
        var w = 675
        var h = 600
        var scale = this.props.width / w

        return (
            <div style={{transformOrigin:" 0 0 0 ",transform:`scale(${scale})`}}>

            {this.state.dynamic && 
            <div style={{display:"flex",
            
            flexWrap:"wrap", 
            color:"#264D8F", 
            alignItems:"center",
            justifyContent:"center", 
            border:"0px dashed red", width:w}}>
            <div style={{textAlign:"center", overflow:"hidden", 
            flexGrow:"none",flexShrink:"none",fontFamily: "'News Cycle', sans-serif",
            lineHeight:"80px",border:"0px dashed yellow",width:w, height:"78px"}}>
                <div ref={this.topText}>
                    <span style={{textTransform:"uppercase",fontSize:"40px"}} >the&nbsp;</span>
                    <span style={{textTransform:"uppercase",fontSize:"80px"}}>Periodic Table</span>
                    <span style={{textTransform:"uppercase",fontSize:"40px"}} >&nbsp;of</span>
                </div>            
            </div>
            <div style={{textAlign:"center", 
                        padding:"10px", 
                        lineHeight:`80px`, 
                        backgroundColor:"#264D8F",
                        color:"#ffffff",
                        flexGrow:"none",flexShrink:"none",
                        width:w,
                        fontSize:"80px"}}>
            {this.state.text}
            {this.props.subtitle && 
            <p style={{fontSize:"30px"}}>
                {this.props.subtitle}
            </p>}
            </div>
            </div>
            }
            {!this.state.dynamic && 
            <div style={{marginTop:"10px",width:"675px",height:"600",border:this.props.border ?this.props.border :  "1px solid #cccccc", overflow:"hide", 
            backgroundColor: this.props.backgroundColor ? this.props.backgroundColor : "#ffffff" ,
            color: this.props.textColor ? this.props.textColor  : "#000000"}} >
            
            <div style={{padding:"10px",textAlign:"center"}} className="ms-font-xl" >
             {this.props.bannerTeaser ? this.props.bannerTeaser : "Entrypoint to the modern workspace" } 
            </div>
            
             <div style={{textAlign:"center"}} >
            <img style={{ height:"auto", maxHeight:"120px",maxWidth:"675px"}} src={this.props.image ?this.props.image : "https://jumpto365.com/resources/images/app/Title-EN.png"}></img>
            
            </div>
           

                        
            <div style={{padding:"10px",textAlign:"center"}} className="ms-font-l" >
            <div>
            {this.props.zoneText ? this.props.zoneText : ""}
            </div>
            
            <div>
            
           
                {this.props.bannerByline ? this.props.bannerByline : "Table designed by Matt Wade, jumpto365, Inc. @thatmattwade"}
                
            </div>
            </div>
            </div>
            }
            {!this.props.hideOurLogo && false && 
            <div style={{textAlign:"center", width:w}} >
            <a href="https://medium.com/jumpto365/latest/home" target="_blank">
            <img src={ jumpto365Logo} style={{width:"auto",height:"48px",marginTop:"15px"}}/> </a>
            </div>
            }
            </div>
        )
    }
}

