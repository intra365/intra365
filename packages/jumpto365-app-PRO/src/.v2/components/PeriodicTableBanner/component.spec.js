import React, { Component } from 'react';

import PeriodicTableBanner from '.'

test('Check that it renders', () => {

    const wrapper = shallow(<PeriodicTableBanner about="🍕"/>);

    expect(wrapper).toMatchSnapshot();
 
}

)