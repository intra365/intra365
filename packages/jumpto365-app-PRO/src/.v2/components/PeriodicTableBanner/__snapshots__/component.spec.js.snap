// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PeriodicTableBanner
    about="🍕"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        style={
                Object {
                        "alignItems": "center",
                        "border": "0px dashed red",
                        "color": "#264D8F",
                        "display": "flex",
                        "flexWrap": "wrap",
                        "justifyContent": "center",
                        "width": 675,
                      }
        }
>
        <div
                style={
                        Object {
                                "border": "0px dashed yellow",
                                "flexGrow": "none",
                                "flexShrink": "none",
                                "fontFamily": "'News Cycle', sans-serif",
                                "height": "78px",
                                "lineHeight": "80px",
                                "overflow": "hidden",
                                "textAlign": "center",
                                "width": 675,
                              }
                }
        >
                <div>
                        <span
                                style={
                                        Object {
                                                "fontSize": "40px",
                                                "textTransform": "uppercase",
                                              }
                                }
                        >
                                the 
                        </span>
                        <span
                                style={
                                        Object {
                                                "fontSize": "80px",
                                                "textTransform": "uppercase",
                                              }
                                }
                        >
                                Periodic Table
                        </span>
                        <span
                                style={
                                        Object {
                                                "fontSize": "40px",
                                                "textTransform": "uppercase",
                                              }
                                }
                        >
                                 of
                        </span>
                </div>
        </div>
        <div
                style={
                        Object {
                                "backgroundColor": "#264D8F",
                                "color": "#ffffff",
                                "flexGrow": "none",
                                "flexShrink": "none",
                                "fontSize": "80px",
                                "lineHeight": "80px",
                                "padding": "10px",
                                "textAlign": "center",
                                "width": 675,
                              }
                }
        >
                Office 365
        </div>
        <a
                href="https://medium.com/jumpto365/latest/home"
                target="_blank"
        >
                <img
                        src="Logo horizontal color - transparent background.png"
                        style={
                                Object {
                                        "height": "48px",
                                        "marginTop": "15px",
                                        "width": "auto",
                                      }
                        }
                />
                 
        </a>
</div>,
      "style": Object {
        "transform": "scale(NaN)",
        "transformOrigin": " 0 0 0 ",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
                        Object {
                                    "border": "0px dashed yellow",
                                    "flexGrow": "none",
                                    "flexShrink": "none",
                                    "fontFamily": "'News Cycle', sans-serif",
                                    "height": "78px",
                                    "lineHeight": "80px",
                                    "overflow": "hidden",
                                    "textAlign": "center",
                                    "width": 675,
                                  }
            }
>
            <div>
                        <span
                                    style={
                                                Object {
                                                            "fontSize": "40px",
                                                            "textTransform": "uppercase",
                                                          }
                                    }
                        >
                                    the 
                        </span>
                        <span
                                    style={
                                                Object {
                                                            "fontSize": "80px",
                                                            "textTransform": "uppercase",
                                                          }
                                    }
                        >
                                    Periodic Table
                        </span>
                        <span
                                    style={
                                                Object {
                                                            "fontSize": "40px",
                                                            "textTransform": "uppercase",
                                                          }
                                    }
                        >
                                     of
                        </span>
            </div>
</div>,
          <div
            style={
                        Object {
                                    "backgroundColor": "#264D8F",
                                    "color": "#ffffff",
                                    "flexGrow": "none",
                                    "flexShrink": "none",
                                    "fontSize": "80px",
                                    "lineHeight": "80px",
                                    "padding": "10px",
                                    "textAlign": "center",
                                    "width": 675,
                                  }
            }
>
            Office 365
</div>,
          <a
            href="https://medium.com/jumpto365/latest/home"
            target="_blank"
>
            <img
                        src="Logo horizontal color - transparent background.png"
                        style={
                                    Object {
                                                "height": "48px",
                                                "marginTop": "15px",
                                                "width": "auto",
                                              }
                        }
            />
             
</a>,
        ],
        "style": Object {
          "alignItems": "center",
          "border": "0px dashed red",
          "color": "#264D8F",
          "display": "flex",
          "flexWrap": "wrap",
          "justifyContent": "center",
          "width": 675,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              <span
                            style={
                                          Object {
                                                        "fontSize": "40px",
                                                        "textTransform": "uppercase",
                                                      }
                            }
              >
                            the 
              </span>
              <span
                            style={
                                          Object {
                                                        "fontSize": "80px",
                                                        "textTransform": "uppercase",
                                                      }
                            }
              >
                            Periodic Table
              </span>
              <span
                            style={
                                          Object {
                                                        "fontSize": "40px",
                                                        "textTransform": "uppercase",
                                                      }
                            }
              >
                             of
              </span>
</div>,
            "style": Object {
              "border": "0px dashed yellow",
              "flexGrow": "none",
              "flexShrink": "none",
              "fontFamily": "'News Cycle', sans-serif",
              "height": "78px",
              "lineHeight": "80px",
              "overflow": "hidden",
              "textAlign": "center",
              "width": 675,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  style={
                                    Object {
                                                      "fontSize": "40px",
                                                      "textTransform": "uppercase",
                                                    }
                  }
>
                  the 
</span>,
                <span
                  style={
                                    Object {
                                                      "fontSize": "80px",
                                                      "textTransform": "uppercase",
                                                    }
                  }
>
                  Periodic Table
</span>,
                <span
                  style={
                                    Object {
                                                      "fontSize": "40px",
                                                      "textTransform": "uppercase",
                                                    }
                  }
>
                   of
</span>,
              ],
            },
            "ref": Object {
              "current": null,
            },
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "the ",
                  "style": Object {
                    "fontSize": "40px",
                    "textTransform": "uppercase",
                  },
                },
                "ref": null,
                "rendered": "the ",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Periodic Table",
                  "style": Object {
                    "fontSize": "80px",
                    "textTransform": "uppercase",
                  },
                },
                "ref": null,
                "rendered": "Periodic Table",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " of",
                  "style": Object {
                    "fontSize": "40px",
                    "textTransform": "uppercase",
                  },
                },
                "ref": null,
                "rendered": " of",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Office 365",
              undefined,
            ],
            "style": Object {
              "backgroundColor": "#264D8F",
              "color": "#ffffff",
              "flexGrow": "none",
              "flexShrink": "none",
              "fontSize": "80px",
              "lineHeight": "80px",
              "padding": "10px",
              "textAlign": "center",
              "width": 675,
            },
          },
          "ref": null,
          "rendered": Array [
            "Office 365",
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                src="Logo horizontal color - transparent background.png"
                style={
                                Object {
                                                "height": "48px",
                                                "marginTop": "15px",
                                                "width": "auto",
                                              }
                }
/>,
              " ",
            ],
            "href": "https://medium.com/jumpto365/latest/home",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": "Logo horizontal color - transparent background.png",
                "style": Object {
                  "height": "48px",
                  "marginTop": "15px",
                  "width": "auto",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            " ",
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          style={
                    Object {
                              "alignItems": "center",
                              "border": "0px dashed red",
                              "color": "#264D8F",
                              "display": "flex",
                              "flexWrap": "wrap",
                              "justifyContent": "center",
                              "width": 675,
                            }
          }
>
          <div
                    style={
                              Object {
                                        "border": "0px dashed yellow",
                                        "flexGrow": "none",
                                        "flexShrink": "none",
                                        "fontFamily": "'News Cycle', sans-serif",
                                        "height": "78px",
                                        "lineHeight": "80px",
                                        "overflow": "hidden",
                                        "textAlign": "center",
                                        "width": 675,
                                      }
                    }
          >
                    <div>
                              <span
                                        style={
                                                  Object {
                                                            "fontSize": "40px",
                                                            "textTransform": "uppercase",
                                                          }
                                        }
                              >
                                        the 
                              </span>
                              <span
                                        style={
                                                  Object {
                                                            "fontSize": "80px",
                                                            "textTransform": "uppercase",
                                                          }
                                        }
                              >
                                        Periodic Table
                              </span>
                              <span
                                        style={
                                                  Object {
                                                            "fontSize": "40px",
                                                            "textTransform": "uppercase",
                                                          }
                                        }
                              >
                                         of
                              </span>
                    </div>
          </div>
          <div
                    style={
                              Object {
                                        "backgroundColor": "#264D8F",
                                        "color": "#ffffff",
                                        "flexGrow": "none",
                                        "flexShrink": "none",
                                        "fontSize": "80px",
                                        "lineHeight": "80px",
                                        "padding": "10px",
                                        "textAlign": "center",
                                        "width": 675,
                                      }
                    }
          >
                    Office 365
          </div>
          <a
                    href="https://medium.com/jumpto365/latest/home"
                    target="_blank"
          >
                    <img
                              src="Logo horizontal color - transparent background.png"
                              style={
                                        Object {
                                                  "height": "48px",
                                                  "marginTop": "15px",
                                                  "width": "auto",
                                                }
                              }
                    />
                     
          </a>
</div>,
        "style": Object {
          "transform": "scale(NaN)",
          "transformOrigin": " 0 0 0 ",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              style={
                            Object {
                                          "border": "0px dashed yellow",
                                          "flexGrow": "none",
                                          "flexShrink": "none",
                                          "fontFamily": "'News Cycle', sans-serif",
                                          "height": "78px",
                                          "lineHeight": "80px",
                                          "overflow": "hidden",
                                          "textAlign": "center",
                                          "width": 675,
                                        }
              }
>
              <div>
                            <span
                                          style={
                                                        Object {
                                                                      "fontSize": "40px",
                                                                      "textTransform": "uppercase",
                                                                    }
                                          }
                            >
                                          the 
                            </span>
                            <span
                                          style={
                                                        Object {
                                                                      "fontSize": "80px",
                                                                      "textTransform": "uppercase",
                                                                    }
                                          }
                            >
                                          Periodic Table
                            </span>
                            <span
                                          style={
                                                        Object {
                                                                      "fontSize": "40px",
                                                                      "textTransform": "uppercase",
                                                                    }
                                          }
                            >
                                           of
                            </span>
              </div>
</div>,
            <div
              style={
                            Object {
                                          "backgroundColor": "#264D8F",
                                          "color": "#ffffff",
                                          "flexGrow": "none",
                                          "flexShrink": "none",
                                          "fontSize": "80px",
                                          "lineHeight": "80px",
                                          "padding": "10px",
                                          "textAlign": "center",
                                          "width": 675,
                                        }
              }
>
              Office 365
</div>,
            <a
              href="https://medium.com/jumpto365/latest/home"
              target="_blank"
>
              <img
                            src="Logo horizontal color - transparent background.png"
                            style={
                                          Object {
                                                        "height": "48px",
                                                        "marginTop": "15px",
                                                        "width": "auto",
                                                      }
                            }
              />
               
</a>,
          ],
          "style": Object {
            "alignItems": "center",
            "border": "0px dashed red",
            "color": "#264D8F",
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "width": 675,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                <span
                                style={
                                                Object {
                                                                "fontSize": "40px",
                                                                "textTransform": "uppercase",
                                                              }
                                }
                >
                                the 
                </span>
                <span
                                style={
                                                Object {
                                                                "fontSize": "80px",
                                                                "textTransform": "uppercase",
                                                              }
                                }
                >
                                Periodic Table
                </span>
                <span
                                style={
                                                Object {
                                                                "fontSize": "40px",
                                                                "textTransform": "uppercase",
                                                              }
                                }
                >
                                 of
                </span>
</div>,
              "style": Object {
                "border": "0px dashed yellow",
                "flexGrow": "none",
                "flexShrink": "none",
                "fontFamily": "'News Cycle', sans-serif",
                "height": "78px",
                "lineHeight": "80px",
                "overflow": "hidden",
                "textAlign": "center",
                "width": 675,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    style={
                                        Object {
                                                            "fontSize": "40px",
                                                            "textTransform": "uppercase",
                                                          }
                    }
>
                    the 
</span>,
                  <span
                    style={
                                        Object {
                                                            "fontSize": "80px",
                                                            "textTransform": "uppercase",
                                                          }
                    }
>
                    Periodic Table
</span>,
                  <span
                    style={
                                        Object {
                                                            "fontSize": "40px",
                                                            "textTransform": "uppercase",
                                                          }
                    }
>
                     of
</span>,
                ],
              },
              "ref": Object {
                "current": null,
              },
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "the ",
                    "style": Object {
                      "fontSize": "40px",
                      "textTransform": "uppercase",
                    },
                  },
                  "ref": null,
                  "rendered": "the ",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Periodic Table",
                    "style": Object {
                      "fontSize": "80px",
                      "textTransform": "uppercase",
                    },
                  },
                  "ref": null,
                  "rendered": "Periodic Table",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " of",
                    "style": Object {
                      "fontSize": "40px",
                      "textTransform": "uppercase",
                    },
                  },
                  "ref": null,
                  "rendered": " of",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Office 365",
                undefined,
              ],
              "style": Object {
                "backgroundColor": "#264D8F",
                "color": "#ffffff",
                "flexGrow": "none",
                "flexShrink": "none",
                "fontSize": "80px",
                "lineHeight": "80px",
                "padding": "10px",
                "textAlign": "center",
                "width": 675,
              },
            },
            "ref": null,
            "rendered": Array [
              "Office 365",
              undefined,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  src="Logo horizontal color - transparent background.png"
                  style={
                                    Object {
                                                      "height": "48px",
                                                      "marginTop": "15px",
                                                      "width": "auto",
                                                    }
                  }
/>,
                " ",
              ],
              "href": "https://medium.com/jumpto365/latest/home",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "Logo horizontal color - transparent background.png",
                  "style": Object {
                    "height": "48px",
                    "marginTop": "15px",
                    "width": "auto",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              " ",
            ],
            "type": "a",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
